cmd/kiam/iptables.go

```diff
@@ -59,13 +61,32 @@ func (r *rules) ruleSpec() []string {
 	return rules
 }
 
+var (
+	retryInterval = time.Millisecond * 500
+	maxAttempts   = 30
+)
+
 func (r *rules) Remove() error {
 	ipt, err := iptables.New()
 	if err != nil {
 		return err
 	}
 
-	return ipt.Delete("nat", "PREROUTING", r.ruleSpec()...)
+	var attempt int
+	for {
+		if attempt >= maxAttempts {
+			log.Errorf("failed to remove iptables rule, retries exhausted: %s", err.Error())
+			break
+		}
+		if err := ipt.Delete("nat", "PREROUTING", r.ruleSpec()...); err == nil {
+			log.Info("iptables rule was removed", err.Error())
```

Comment:
why have the error here?  we've already checked that err is nil

Reply:
Removed it, thanks